### PR TITLE
Add --json flag and default to plain text

### PR DESCRIPTION
**Pull request by gimenete**
We want to always defualt to plain text ouput. For most commands we want to also support the `--json` flag to return the output in JSON format, which will allow people to automate things in scripts and such.

I'm putting this PR in _ready for review_ but the plain text output needs to be polished. The reason to ask for reviews is to get reviews about the general approach. 🙏

### CODE REVIEW

File: cmd/branches.go [hunk 1]
```diff
@@ -53,7 +53,11 @@ func listBranches(c *cli.Context) error {
 	}
 
 	resp, err := client.GetBranchList(c.Context, spec.DBNameParam(dbName))
-	return printResponse(c, resp, err)
+	return printResponse(c, resp, err, printBranches)
+}
+
+func printBranches(response map[string]interface{}) {
+	fmt.Println("print branches!!", response)
```
Comment (gimenete) [inline cmd/branches.go:60]: What would be the best way to navigate the response to loop and access specific values? Using `reflect`? Casting to manually written types? Should we auto generate Go code form the open api spec?

Comment (tsg) [reply cmd/branches.go:60]: It does generate types for responses, but you need to use the `WithResponses` versions of the clients and functions. See an example here: https://github.com/xataio/cli/blob/main/cmd/branches.go#L151:L151

I find the generated code pretty clunky so I'm hoping we're going to find a better generator, but it should work here.

If you need to keep the `map[string]interface{}` the standard way to "walk" that is via [type switches](https://go.dev/doc/effective_go#type_switch).

Comment (gimenete) [reply cmd/branches.go]: Good to know! I've pushed some changes using the `WithResponse` versions. Would love to know what you think!

File: cmd/branches.go [hunk 2]
```diff
@@ -52,8 +52,22 @@ func listBranches(c *cli.Context) error {
...
+			return fmt.Errorf("Unexpected server response %s", resp.Status())
+		}
+		fmt.Println("Branches for database", data.DatabaseName)
+		fmt.Println()
+
+		for i := 0; i < len(data.Branches); i++ {
+			branch := data.Branches[i]
+			fmt.Println("- ", branch.Name)
+		}
+		return nil
```
Comment (exekias) [inline cmd/branches.go:69]: I wonder if we should standardize this, ie having a way to print lists or tables from a given input. Doing it in a custom function means we need to update all of them if we change format.

Ideally, this `printResponse` function could take the list of items to print, together with the raw json, then decide the output format based on flags.

Comment (gimenete) [reply cmd/branches.go:69]: That's a good idea. I'm still thinking on how to present all the data. So far I'm just printing names, but we will need to print created at dates, maybe the author/creator/owner of a branch/workspace,... Maybe we can present the data in a tabular format? Not a heavy weight tabular format, but something like when you do `ls -o` for example. That could be standarized easily passing an array of values/columns for each item in a list.

Comment (exekias) [reply cmd/branches.go:69]: +1 to tabular/table format, the nice way about organizing the code around formatters/printers is that we can iterate over them any time. This is how `kubectl` handles output, for inspiration: https://pkg.go.dev/k8s.io/cli-runtime/pkg/printers. Also `gh` uses a table printer like this: https://github.com/cli/cli/blob/8dbd07212c239d32675d277363cc1f492ab62ad0/pkg/cmd/pr/list/list.go#L196

Comment (gimenete) [reply cmd/branches.go]: I've just implemented a `printTable` utility https://github.com/xataio/cli/pull/57/commits/27338f3a90a0a9857090b64b2803cb0b3ccffc1d

It uses the built-in `text/tabwriter` module, and does some useful things such as printing dates.

This is how it looks with the branches list command:

image

**Review by exekias — APPROVED** [review]

**Pull request merged**